support/cas-server-support-pac4j-core-clients/src/main/java/org/apereo/cas/support/pac4j/web/flow/SingleLogoutPreparationAction.java

```diff
@@ -0,0 +1,85 @@
+package org.apereo.cas.support.pac4j.web.flow;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apereo.cas.util.Pac4jUtils;
+import org.apereo.cas.web.support.CookieRetrievingCookieGenerator;
+import org.apereo.cas.web.support.WebUtils;
+import org.pac4j.core.context.J2EContext;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.ProfileManager;
+import org.pac4j.core.profile.service.ProfileService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.webflow.action.AbstractAction;
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.execution.RequestContext;
+
+
+/**
+ * The purpose of this action is to prepare the PAC4J Profile Manager for Single Logout.
+ * 
+ * The Profile Manager keeps the profiles in request + session but the session has already been destroyed. This action should restore the
+ * profile from a long term storage - {@link ProfileService} and populate the PAC4J Profile Manager with it.
+ * 
+ * This action should be called from the Logout web flow.
+ * 
+ * @author jkacer
+ * 
+ * @since 5.3.0
+ */
+public class SingleLogoutPreparationAction extends AbstractAction {
+
+    /** ID of the web flow state where the action will be inserted. */
+    public static final String WEBFLOW_ACTION_STATE_ID = "prepareForSingleLogout";
+
+    /** Evaluation expression for the web flow state where the action will be called. */
+    public static final String WEBFLOW_ACTION_EVAL_EXPRESSION = "pac4jSingleLogoutPreparationAction";
+
+    private final Logger logger2 = LoggerFactory.getLogger(SingleLogoutPreparationAction.class);
```

Comment:
loggers need to be static and final.

Reply:
Changed to static final LOGGER everywhere. Where needed, I used name LOGGER2 because my internal Sonar was complaining about a name clash with the parent class's logger.